# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

Makes M4 not use the gnulib freadahead function which has a non-portable
implementation and isn't needed.

diff --git lib/closein.c lib/closein.c
index 7415b19..6f5b574 100644
--- lib/closein.c
+++ lib/closein.c
@@ -31,7 +31,6 @@
 #include "closeout.h"
 #include "error.h"
 #include "exitfail.h"
-#include "freadahead.h"
 #include "quotearg.h"
 
 static const char *file_name;
@@ -80,17 +79,6 @@ close_stdin (void)
 {
   bool fail = false;
 
-  /* There is no need to flush stdin if we can determine quickly that stdin's
-     input buffer is empty; in this case we know that if stdin is seekable,
-     (fseeko (stdin, 0, SEEK_CUR), ftello (stdin))
-     == lseek (0, 0, SEEK_CUR).  */
-  if (freadahead (stdin) > 0)
-    {
-      /* Only attempt flush if stdin is seekable, as fflush is entitled to
-         fail on non-seekable streams.  */
-      if (fseeko (stdin, 0, SEEK_CUR) == 0 && fflush (stdin) != 0)
-        fail = true;
-    }
   if (close_stream (stdin) != 0)
     fail = true;
   if (fail)

